tools/freeze_requirements.py

```diff
@@ -29,8 +31,47 @@ def worker(args: tuple[str, ...]) -> None:
         )
 
 
-def main() -> int:
+# XXX(typing): it's supposed to be list[Future]
+def check_futures(futures: list[Any]) -> int:
```

Comment:
should this be `list[concurrent.futures.Future[subprocess.CompletedProcess[bytes]]]` ?

Reply:
Ah yeah, but I don't think it likes the potentially empty list?

```
tools/freeze_requirements.py:153: error: Argument 1 to "check_futures" has incompatible type "List[Future[None]]"; expected "List[Future[CompletedProcess[bytes]]]"
```

Reply:
oh actually it'll be `list[concurrent.futures.Future[None]]` -- I misread the definition of `worker` above